source/nijigenerate/io/package.d

```diff
@@ -166,19 +153,35 @@ string incShowSaveDialog(const(TFD_Filter)[] filters, string fname, string title
 
             // FALLBACK: If xdg-desktop-portal is not available then try tinyfiledialogs.
             c_str filename = tinyfd_saveFileDialog(title.toStringz, fname.toStringz, filters);
-            if (filename !is null) {
-                string file = cast(string) filename.fromStringz;
-                return file;
-            }
-            return null;
+            return incToDString(filename);
         }
     } else {
         c_str filename = tinyfd_saveFileDialog(title.toStringz, fname.toStringz, filters);
-        if (filename !is null) {
-            string file = cast(string) filename.fromStringz;
-            return file;
+        return incToDString(filename);
+    }
+}
+
+string incMessageBox(string title, string message, string dialogType = "ok", string iconType = "info", int defaultButton = 0) {
```

Comment:
can you define enum instead of string?

e.g.

enum AskDialogAnswer {
   Cancel, Yes, No
}